src/app/core/modals/platform-credentials-modal/platform-credentials-modal.component.ts

```diff
@@ -136,14 +139,23 @@ export class PlatformCredentialsModalComponent implements OnInit {
 
                 let maybeUserUpdate = Promise.resolve();
 
-                if (isEditing && editedCredentials.equals(active)) {
-                    // If we are editing credentials that appear to be active, update it
-                    maybeUserUpdate = this.auth.setActiveCredentials(editedCredentials);
+                if (isEditing) {
+                    if (editedCredentials.equals(active)) {
+                        // If we are editing credentials that appear to be active, update it
+                        maybeUserUpdate = this.auth.setActiveCredentials(editedCredentials);
+                    }
 
-                } else if (all.length === 1) {
-                    // Otherwise, if we added new credentials, and it turns out that it's the first one,
-                    // activate that user
-                    maybeUserUpdate = this.auth.setActiveCredentials(all[0]);
+                } else {
+                    // Activate added credentials
+                    maybeUserUpdate = this.auth.setActiveCredentials(credentials);
+                    this.notificationBarService.showNotification(GetStartedNotificationComponent, {
+                        type: "info",
+                        componentInputs: {
```

Comment:
Why not return the notification instance like ModalService does, then set inputs/outputs on the instance?